src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/ILspLogger.cs

```diff
@@ -10,8 +10,13 @@
 namespace Microsoft.CommonLanguageServerProtocol.Framework;
 internal interface ILspLogger
 {
+    [Obsolete($"Use {nameof(CreateContext)} instead.")]
```

Comment:
If it's safe for me to delete these methods (because CLaSP is a source package), I'm happy to do so. The replacement is to set a scope, and then fire a start/end message inside of it.